Cabal/Distribution/Simple/Register.hs

```diff
@@ -230,13 +232,23 @@ invokeHcPkg verbosity comp conf dbStack extraArgs =
 withHcPkg :: String -> Compiler -> ProgramConfiguration
           -> (HcPkgInfo -> IO a) -> IO a
 withHcPkg name comp conf f =
-  case compilerFlavor comp of
-    GHC   -> f (GHC.hcPkgInfo conf)
-    GHCJS -> f (GHCJS.hcPkgInfo conf)
-    LHC   -> f (LHC.hcPkgInfo conf)
+  case getHcPkgInfo comp conf of
+    Just hcPkgInfo -> f hcPkgInfo
     _     -> die ("Distribution.Simple.Register." ++ name ++ ":\
                   \not implemented for this compiler")
 
+getHcPkgInfo :: Compiler -> ProgramConfiguration -> Maybe HcPkgInfo
+getHcPkgInfo comp conf = case compilerFlavor comp of
+                           GHC   -> Just (GHC.hcPkgInfo conf)
+                           GHCJS -> Just (GHCJS.hcPkgInfo conf)
+                           LHC   -> Just (LHC.hcPkgInfo conf)
+                           _     -> Nothing
+
+multInstEnabled :: Compiler -> ProgramConfiguration -> Bool
```

Comment:
Sorry, I overlooked the line you had commented and thought that you were referring to `getHcPkgInfo`. But I looked into `D.S.Compiler` and it seems to be focused only on compiler features with `ghc --info` and `multInstEnabled` is a ghc-pkg feature. Also `D.S.Compiler` is imported by `D.S.Programs.HcPkg` now and so `HcPkgInfo` cannot be available to `D.S.Compiler` unless we move things from one module to another.